lib/oban/job.ex

```diff
@@ -643,4 +642,22 @@ defmodule Oban.Job do
         {:error, "unknown option, #{inspect(option)}"}
     end)
   end
+
+  defp validate_keys(keys, unique) do
+    cond do
+      keys == [] ->
+        :ok
+
+      not (is_list(keys) and Enum.all?(keys, &is_atom/1)) ->
+        {:error, "expected :keys to be a list of atoms"}
+
+      not (is_list(keys) and
+               Enum.any?(@key_compatible_fields, &(&1 in Keyword.get(unique, :fields)))) ->
```

Comment:
Changed for readability and conciseness. 

```suggestion
      is_nil(keys) and Enum.any?(@keyable_fields, &(&1 in Keyword.get(unique, :fields)))) ->
```